Core/Src/porting/lib/lzma/7zTypes.h

```diff
@@ -0,0 +1,376 @@
+/* 7zTypes.h -- Basic types
+2018-08-04 : Igor Pavlov : Public domain */

```

Comment:
Could you add a source to where you downloaded the third party files from? There are quite a few copies floating around so it would be nice to link to the source, just so there won't be any questions about the Public Domain claim.